### DISCUSSION TITLE
Hydrogen

### DISCUSSION THREAD

**Discussion by brucebastian** [pulsar-edit/.github#232] [Community Package Help]
ATOM + Hydrogen was the go-to REPL environment for many data scientists and data analysts.  I have tried a number of IDEs and it's embedded cells are not only unique, but for me, nothing else I've  tried is quite as nice.  But it is extremely hard to get working with Pulsar for someone who is not a computer scientist and despite following this guide https://pipegalera.com/posts/pulsar-hydrogen/, I am yet to get it working on Windows 10.

Can someone who is more knowledgeable than me fork it or update it so that it can be installed via the Pulsar package manager.  I would not be alone in this.  Hydrogen is a very popular package (over 2.5 million downloads) and it's embedded cells are unique.

**Comment by savetheclocktower** [comment]
I can point out one thing that won't work on Windows from that guide: all references to `pulsar -p` only work on macOS and Linux. That first example from the post…

```
pulsar -p install https://github.com/nteract/hydrogen -t v2.16.5
```

…should instead be this on Windows…

```
ppm install https://github.com/nteract/hydrogen -t v2.16.5
```

I'm sure there are other issues you may run into, so if you're still having trouble, I'd suggest [joining our Discord](https://discord.gg/7aEbB9dGRT) so that someone can help you troubleshoot in real time. Apologies for the troubles, but Hydrogen really is one of the most complex packages in Pulsar.

**Comment by brucebastian** [comment]
Thanks so much for this Andrew, that's brilliant.

I will make these changes, give it a shot, then try discord for more help.

I really don't know what I'm doing, but I do love Hydrogen, so if I can get
it working with help, I think that will be a tremendous help to other
Windows users.

Appreciate your help so far.

Cheers,
Bruce.

**Reply by savetheclocktower** [reply]
No problem. I just noticed you're in Australia, so it might be a slight challenge finding common hours with our team that's mostly in the  UK and the Americas. But we'll give it a shot!
